src/plugins/intel_cpu/tests/unit/snippets_transformations/x64/lowered/buffer_allocation.cpp

```diff
@@ -255,6 +256,11 @@ TEST_P(MHAFP32BufferAllocationTest, BufferAllocationCPU) {
 }
 
 TEST_P(MHABF16AMXBufferAllocationTest, BufferAllocationCPU) {
+    // Scratchpad memory for AMX with CopyA (dynamic case) has allocation size which depends on element count in vector register.
+    // So the current `expected_allocation_size` in the test is targeted on real AVX512 platforms with vector registers with 512 bits.
+    // If the test infrastructure has AVX2, the allocation size will not be matched.
+    if (m_linear_ir.is_dynamic() && !with_cpu_x86_avx512_core())
```

Comment:
It's totally clear from the comment why we need to check ISA here. But why do we need to check `m_linear_ir.is_dynamic()`?